std/socket.d

```diff
@@ -328,33 +342,11 @@ shared static this() @system
         if (val)         // Request Winsock 2.2 for IPv6.
             throw new SocketOSException("Unable to initialize socket library", val);
 
-        // These functions may not be present on older Windows versions.
-        // See the comment in InternetAddress.toHostNameString() for details.
-        auto ws2Lib = GetModuleHandleA("ws2_32.dll");
-        if (ws2Lib)
-        {
-            getnameinfoPointer = cast(typeof(getnameinfoPointer))
-                                 GetProcAddress(ws2Lib, "getnameinfo");
-            getaddrinfoPointer = cast(typeof(getaddrinfoPointer))
-                                 GetProcAddress(ws2Lib, "getaddrinfo");
-            freeaddrinfoPointer = cast(typeof(freeaddrinfoPointer))
-                                 GetProcAddress(ws2Lib, "freeaddrinfo");
-        }
-    }
-    else version(Posix)
-    {
-        getnameinfoPointer = &getnameinfo;
-        getaddrinfoPointer = &getaddrinfo;
-        freeaddrinfoPointer = &freeaddrinfo;
-    }
-}
-
+        // Now that we called WSAStartup, make sure we clean up, too.
+        import core.stdc.stdlib;
```

Comment:
Urgh. Not a fan of this rule. I recall we introduced it to compensate for hijacking, which we since fixed.